app/domain/time_entry/description_rules.rb

```diff
@@ -0,0 +1,38 @@
+class TimeEntry::DescriptionRules
+  JIRA_REGEX = /(?:\s|^)([A-Z]+-[0-9]+)(?=\s|$)/
+
+  def initialize(entry, ruleset = :internal_employee)
+    @description = entry.description
+    @ruleset = ruleset
+  end
+
+  def valid?
+    return internal_employee() if @ruleset == :internal_employee
+    false
+  end
+
+    private
+
+    def has_word_count(length)
+      return true if @description.split.size > length - 1
+      false
+    end
+
+    def has_calls_tag
+      @description.downcase.include?("#calls")
+    end
+
+    def has_url
+      @description.downcase.include?("http")
+    end
+
+    def has_jira_ticket
+      @description.gsub(/[\[\]\,\.]/, ' ').scan(JIRA_REGEX).any?
```

Comment:
maybe we can do `JIRA_REGEX.match?(@description)` ? or there's a reason we are replacing the description instead of just checking if it has a jira ticket?

Reply:
So the JIRA_REGEX is the one provided by Atlassian which assumes a space before the ID. Looking through the database `[XXX-111,YYY-222]` (n Jira ids with no spaces) is a pretty common entry... so I'm subbing square brackets, commas, and periods then scanning for the regex.

I could alter the JIRA_REGEX (but then that will get super complex)... could also throw in another method to make it more clear what's going on?

```
  def has_jira_ticket
    JIRA_REGEX.match?(replace_special_chars(@description))`
  end

  def replace_special_chars(@description)
    @description.gsub(/[\[\]\,\.]/, ' ')
  end
```

Not sure what would be easiest to understand

At a minimum I'll add a comment explaining what's going on.